math/mathcore/inc/Math/LFSR.h

```diff
@@ -0,0 +1,153 @@
+    * @see https://en.wikipedia.org/wiki/Monic_polynomial, https://en.wikipedia.org/wiki/Linear-feedback_shift_register, https://en.wikipedia.org/wiki/Pseudorandom_binary_sequence
+    */
+   template <size_t k, size_t nTaps>
+   static bool
+   NextLFSR(std::bitset<k>& lfsr, const std::array<uint16_t, nTaps> taps, const bool left = true)
+   {
+      static_assert(k <= 32, "For the moment, only supported until k == 32.");
+      static_assert(k > 0, "Non-zero degree is needed for the LFSR.");
+      static_assert(nTaps > 0, "At least one tap is needed for the LFSR.");
+      static_assert(nTaps <= k, "Cannot use more taps than polynomial order");
+
+      // First, calculate the XOR (^) of all selected bits (marked by the taps)
+      bool newBit = lfsr[taps.at(0) - 1]; // the exponent E of the polynomial correspond to index E - 1 in the bitset
+      for(uint16_t j = 1; j < nTaps ; ++j)
+      {
+         newBit ^= lfsr[taps.at(j) - 1];
+      }
+
+      //Apply the shift to the register in the right direction, and overwrite the empty one with newBit
+      if(left)
+      {
+         lfsr <<= 1;
+         lfsr[0] = newBit;
+      }
+      else
+      {
+         lfsr >>= 1;
+         lfsr[k-1] = newBit;
+      }
+
+      return newBit;
+   }
+
+   /**
+    * @brief Generation of a sequence of pseudo-random bits using a linear feedback shift register (LFSR), until a register value is repeated (or maxPeriod is reached)
+    * @tparam k the length of the LFSR, usually also the order of the monic polynomial PRBS-k (last exponent)
+    * @tparam nTaps the number of taps
+    * @param start the start value (seed) of the LFSR
+    * @param taps the taps that will be XOR-ed to calculate the new bit. They are the exponents of the monic polynomial. Ordering is unimportant. Note that an exponent E in the polynom maps to bit index E-1 in the LFSR.
+    * @param left if true, the direction of the register shift is to the left <<, the newBit is set on lfsr at bit position 0 (right). If false, shift is to the right and the newBit is stored at bit position (k-1)
+    * @param wrapping if true, allow repetition of values in the LFSRhistory, until maxPeriod is reached or the repeated value == start. Enabling this option saves memory as no history is kept
+    * @param oppositeBit if true, use the high/low bit of the LFSR to store output (for left=true/false, respectively) instead of the newBit returned by ::NextLFSR
+    * @return the array of pseudo random bits, or an empty array if input was incorrect
+    * @see https://en.wikipedia.org/wiki/Monic_polynomial, https://en.wikipedia.org/wiki/Linear-feedback_shift_register, https://en.wikipedia.org/wiki/Pseudorandom_binary_sequence
+    */
+   template <size_t k, size_t nTaps>
+   static std::vector<bool>
+   GenerateSequence(const std::bitset<k> start, const std::array<uint16_t, nTaps> taps, const bool left = true, const bool wrapping = false, const bool oppositeBit = false)
```

Comment:
```suggestion
   GenerateSequence(const std::bitset<k> start, const std::array<std::uint16_t, nTaps> taps, const bool left = true, const bool wrapping = false, const bool oppositeBit = false)
```
Prefer `std::uint16_t`.